File: client/src/app/_models/userParams.ts
import { User } from './user';

export class UserParams {
  desiredGender: string;
  minAge = 18;
  maxAge = 100;
  // pagination
  pageNumber = 1;
  pageSize = 10;
  // sorting
  orderBy = 'lastActive';

  constructor(user: User) {
    this.desiredGender = user.gender === 'female' ? 'male' : 'female';
  }
}

File: API/Helpers/UserParams.cs
namespace API.Helpers;

public class UserParams : PaginationParams
{
    public string CurrentUsername { get; set; }
    public string DesiredGender { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 100;
    public string OrderBy { get; set; } = "lastActive";
}
